randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -58,18 +115,66 @@ def _start_point_ref_for(self, node: Node) -> dict:
         else:
             return {}
 
+    def _key_error_for_node(self, node: Node, err: KeyError) -> KeyError:
+        return KeyError(f"{self.game.region_list.node_name(node, with_region=True)} has no extra {err}")
+
+    def _level_name_for(self, node: Node) -> str:
+        region = self.game.region_list.nodes_to_region(node)
+        return region.extra["scenario_id"]
+
+    def _teleporter_ref_for(self, node: Node, actor_key: str = "actor_name") -> dict:
+        try:
+            return {
+                "scenario": self._level_name_for(node),
+                "actor": node.extra[actor_key],
+            }
+        except KeyError as e:
+            return {}
+            raise self._key_error_for_node(node, e)
+
     def _pickup_detail_for_target(self, detail: ExportedPickupDetails) -> dict | None:
+        alt_model = _ALTERNATIVE_MODELS.get(detail.model, [detail.model.name])
+        model_names = alt_model
+
+        if detail.other_player:
+            if model_names == ["offworld"]:
+                base_icon = "unknown"
+                model_names = ["itemsphere"]
+            else:
+                base_icon = detail.model.name
+
+            map_icon = {"custom_icon": {"label": detail.name.upper(), "base_icon": base_icon}}
+        elif alt_model[0] == "itemsphere":
+            map_icon = {
+                "custom_icon": {
+                    "label": detail.original_pickup.name.upper(),
+                }
+            }
+        else:
+            map_icon = {"icon_id": detail.model.name}
+
+        resources = get_resources_for_details(detail.original_pickup, detail.conditional_resources, detail.other_player)
+
         pickup_node = self.game.region_list.node_from_pickup_index(detail.index)
         pickup_type = pickup_node.extra.get("pickup_type", "actor")
 
         hud_text = detail.collection_text[0]
         if len(set(detail.collection_text)) > 1:
             hud_text = self.memo_data[detail.original_pickup.name]
 
-        details = {
-            "pickup_type": pickup_type,
-            "caption": hud_text,
-        }
+        details = {"pickup_type": pickup_type, "caption": hud_text, "resources": resources}
+
+        if pickup_type == "actor":
+            pickup_actor = self._teleporter_ref_for(pickup_node)
+
+            if "map_icon_actor" in pickup_node.extra:
+                map_icon.update({"original_actor": self._teleporter_ref_for(pickup_node, "map_icon_actor")})
```

Comment:
It seems this is not a thing in any pickup node, since it's not covered by the tests.